src/main/java/org/opensearch/security/action/tenancy/MultiTenancyConfigurationsRestHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.action.tenancy;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+public class MultiTenancyConfigurationsRestHandler extends BaseRestHandler {
```

Comment:
The name doesn't seem right, rename all these classes to `TenancyConfig` in place of `MultiTenancyConfigurations`.  Its a single configuration and mutli-tenancy is overly verbose.  We shouldn't change the config value for backwards compatibility, but the new classes don't need to follow suite.

Reply:
Note; there are many places where the term 'configurations' is used, please do a pass over the whole PR to clean these up.

Reply:
Cool.